app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
+    let newUserPromises: Array<Promise<APIUserType>>;
+    if (this.state.showOnlySharedExperiences && this.props.selectedUsers.length > 1) {
+      const relevantEntries = this.state.multipleUsersEntries.filter(entry => entry.isShared);
+      newUserPromises = this.props.selectedUsers.map(user => {
+        const newExperiences = { ...user.experiences };
+        relevantEntries.forEach(entry => {
+          newExperiences[entry.domain] = entry.value;
+        });
+        const newUser = { ...user, experiences: newExperiences };
+        return this.sendUserToServer(newUser, user);
+      });
+    } else {
+      const newExperiences = {};
+      if (this.props.selectedUsers.length === 1) {
+        this.state.singleUsersEntries.forEach(entry => {
+          newExperiences[entry.domain] = entry.value;
+        });
+      } else {
+        this.state.multipleUsersEntries.forEach(entry => {
+          newExperiences[entry.domain] = entry.value;
+        });
+      }
+      newUserPromises = this.props.selectedUsers.map(user => {
+        const newUser = { ...user, experiences: newExperiences };
+        return this.sendUserToServer(newUser, user);
       });
-
-      this.closeModal(newUserPromises);
     }
+    this.resolvePromisesAndCloseModal(newUserPromises);
   };
 
-  /**
-   * Save a user object to the server using an API call.
-   * @param newUser - A modified user object intended to be saved.
-   * @param oldUser - The original user object of `newUser`. Returned in case API call fails
-   *
-   */
   sendUserToServer(newUser: APIUserType, oldUser: APIUserType): Promise<APIUserType> {
     return updateUser(newUser).then(() => Promise.resolve(newUser), () => Promise.reject(oldUser));
   }
 
-  closeModal(usersPromises: Array<Promise<APIUserType>>): void {
+  resolvePromisesAndCloseModal(usersPromises: Array<Promise<APIUserType>>): void {
     Promise.all(usersPromises).then(
       newUsers => {
         this.setState({
-          domain: null,
-          level: null,
+          multipleUsersEntries: [],
+          singleUsersEntries: [],
         });
         this.props.onChange(newUsers);
       },
-      () => {
-        // do nothing and keep modal open
+      error => {
+        handleGenericError(error);
       },
     );
   }
 
-  render() {
-    if (!this.props.visible) {
-      return null;
+  recordModifiedAndExistedBefore = (record: SingleUserTableEntry): boolean => {
+    if (this.props.selectedUsers.length === 1)
+      return (
+        record.value !== this.props.selectedUsers[0].experiences[record.domain] &&
+        record.domain in this.props.selectedUsers[0].experiences
+      );
+    else return false;
+  };
+
+  setValueOfEntry = (index: number, value: number) => {
+    if (value > 0) {
+      if (this.props.selectedUsers.length === 1) {
+        this.setState(prevState => ({
+          singleUsersEntries: prevState.singleUsersEntries.map((entry, currentIndex) => {
+            if (currentIndex === index) {
```

Comment:
The function which is passed to `map` is almost (?) the same as in lines 196 - 207. Please extract this into a variable to avoid duplicated code. The same is true for some other methods in this file.

In general, we might want to rethink whether it's a good idea to have the two variables `singleUsersEntries` and `multipleUsersEntries`. Maybe it should just be `userEntries` which is able to deal with both use cases? Would it work to always use `MultipleUserTableEntry`? The `SingleUserTableEntry` is only a subset of `MultipleUserTableEntry` anyway, right? So, we wouldn't lose anything if we always use the `MultipleUserTableEntries` type?